syntax = "proto3";

package bayesmix;

import "matrix.proto";

message DPState {
  double totalmass = 1;
}

message PYState {
  double strength = 1;
  double discount = 2;
}

message LogSBState {
  Vector regression_coeffs = 1;
}

message MixingState {
  oneof state {
    DPState dp_state = 1;
    PYState py_state = 2;
    LogSBState log_sb_state = 3;
  }
}
